### PR TITLE
lib, test: add session test

### PR DESCRIPTION
**Pull request by nechaido**

### CODE REVIEW

File: test/utils/session/session-serde.js [hunk 1]
```diff
@@ -12,14 +12,13 @@ function stringifySession(session) {
   return JSON.stringify(copy);
 }
 
-function parseSession(sessionString) {
-  const sessionCopy = JSON.parse(sessionString);
-  sessionCopy.buffer = new Map(sessionCopy.buffer);
-  return Object.assign(new jstp.Session(), sessionCopy);
+function deserialize(string) {
```
Comment (belochub) [inline test/utils/session/session-serde.js]: I don't like having `string` as an argument name, can we maybe consider changing it to `sessionCopy` or just `session`?

File: test/node/session-resend-event-on-connection-drop.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+};
+
+const application = new jstp.Application('testApp', ifaces);
+const serverConfig = { applications: [application] };
+const server = jstp.net.createServer(serverConfig);
+
+client.on('message', ([message, ...args]) => {
+  switch (message) {
+    case 'error':
+      test.fail('must not encounter an error');
```
Comment (belochub) [inline test/node/session-resend-event-on-connection-drop.js]: You probably forgot to also print the error you are passing in the message.

File: test/node/session-resend-event-on-connection-drop.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+});
+
+server.listen(0, () => {
+  client.send(['connect', server.address().port]);
+});
+
+function getClient() {
+  return cp.fork(
+    path.join(__dirname, '../utils/session/client'),
+    [],
```
Comment (belochub) [inline test/node/session-resend-event-on-connection-drop.js]: You can omit the `args` argument, according to the [docs](https://nodejs.org/dist/latest-v6.x/docs/api/child_process.html#child_process_child_process_fork_modulepath_args_options) it is optional.

File: test/utils/session/session-serde.js (hunk 1 above)
Comment (nechaido) [reply test/utils/session/session-serde.js]: What about `serializedSession`?

File: test/node/session-resend-event-on-connection-drop.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    { stdin: 'inherit', stdout: 'inherit', stderr: 'inherit' }
```
Comment (belochub) [inline test/node/session-resend-event-on-connection-drop.js]: There are no such options, and what you probably wanted to do can be done by setting the `silent` option to `false`, which is its default value in Node.js 6 and Node.js 8, meaning you can completely avoid providing options.

File: test/node/session-resend-event-on-connection-drop.js (hunk 2 above)
Comment (nechaido) [reply test/node/session-resend-event-on-connection-drop.js]: Fixed.

File: test/utils/session/session-serde.js (hunk 1 above)
Comment (belochub) [reply test/utils/session/session-serde.js]: SGTM

File: test/node/session-resend-event-on-connection-drop.js (hunk 2 above)
Comment (belochub) [reply test/node/session-resend-event-on-connection-drop.js]: @nechaido, it hasn't been fixed yet.

**Comment by nechaido** [comment]
Landed in  297c432 and d4710f4.

**Pull request closed without merging**